JustSaying.Messaging/Extensions/TypeExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace JustSaying.Messaging.Extensions
+{
+    public static class TypeExtensions
+    {
+        public static string ToKey(this Type type)
+        {
+            return string.Join("_", type.Flatten().Select(t => Regex.Replace(t.Name + "_" + t.Namespace, "\\W", "_"))).TruncateTo(100);
```

Comment:
@SamHWhale - This is problematic for us because it also impacts non-generic message types. This is going to break our existing conventions for naming topics.